core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -317,6 +333,38 @@ public UpdateSchema unionByNameWith(Schema newSchema) {
     return this;
   }
 
+  @Override
+  public UpdateSchema addIdentifierField(String name) {
+    Types.NestedField field = schema.findField(name);
+    if (field == null) {
+      field = adds.get(TABLE_ROOT_ID).stream()
+          .filter(f -> f.name().equals(name))
+          .findAny().orElse(null);
+    }
+
+    Preconditions.checkArgument(field != null,
+        "Cannot find column of the given name in existing or newly added columns: %s", name);
+    Preconditions.checkArgument(field.isRequired(),
+        "Cannot add column %s as an identifier field because it is not a required field", name);
+    Preconditions.checkArgument(field.type().isPrimitiveType(),
+        "Cannot add column %s as an identifier field because it is not a primitive field", name);
```

Comment:
We normally omit the "because ..." and use `:` instead. Like this: `Cannot add %s as an identifier field: not a primitive` or `Cannot add %s as an identifier field: already an identifier field`.